polyfill/lib/monthday.mjs

```diff
@@ -65,19 +82,21 @@ export class MonthDay {
   withYear(item) {
     if (!ES.IsTemporalMonthDay(this)) throw new TypeError('invalid receiver');
     let year;
-    if (typeof item === 'object') {
+    if (typeof item === 'object' && item !== null) {
       ({ year } = ES.ToRecord(item, [['year']]));
     } else {
       year = ES.ToInteger(item);
     }
-    const month = GetSlot(this, ISO_MONTH);
-    const day = GetSlot(this, ISO_DAY);
+    const calendar = GetSlot(this, CALENDAR);
+    const day = calendar.day(this);
+    const month = calendar.month(this);
     const Date = GetIntrinsic('%Temporal.Date%');
-    return new Date(year, month, day);
+    return calendar.dateFromFields({ year, month, day }, { disambiguation: 'reject' }, Date);
```

Comment:
Remember to handle an object argument like `monthDay.withYear({ era: "reiwa", year: 2 })`

Reply:
That code was in my Japanese calendar branch, I'll make sure to bring it into this one